src/banded/bidiagonalconjugation.jl

```diff
@@ -0,0 +1,141 @@
+    cᵢᵢ, cᵢ₊₁ᵢ = C[i, i], C[i+1, i]
+    Uᵢ⁻¹ = inv(uᵢᵢ * uᵢ₊₁ᵢ₊₁ - uᵢᵢ₊₁ * uᵢ₊₁ᵢ)
+    dv[i] = Uᵢ⁻¹ * (uᵢ₊₁ᵢ₊₁ * cᵢᵢ - uᵢᵢ₊₁ * cᵢ₊₁ᵢ)
+    ev[i] = Uᵢ⁻¹ * (uᵢᵢ * cᵢ₊₁ᵢ - uᵢ₊₁ᵢ * cᵢᵢ)
+    return data
+end
+
+function _compute_columns!(data::BidiagonalConjugationData, i)
+    ds = data.datasize
+    up = data.uplo == 'U'
+    U, C = data.U, data.C # Treat _compute_column_(up/lo) as function barriers and take these out early
+    return __compute_columns!(data, U, C, i)
+end
+function __compute_columns!(data::BidiagonalConjugationData, U, C, i)
+    ds = data.datasize 
+    up = data.uplo == 'U'
+    for j in (ds+1):i
+        up ? _compute_column_up!(data, U, C, j) : _compute_column_lo!(data, U, C, j)
+    end
+    data.datasize = i
+    return data
+end
+
+function resizedata!(data::BidiagonalConjugationData, n)
+    ds = data.datasize
+    n ≤ ds && return data
+    dv, ev = data.dv, data.ev
+    resize!(dv, 2n + 1)
+    resize!(ev, 2n)
+    return _compute_columns!(data, 2n)
+end
+
+struct BidiagonalConjugationBand{T} <: LazyVector{T}
+    data::BidiagonalConjugationData{T}
+    diag::Bool # true => diagonal, false => offdiagonal 
+end
+@inline size(A::BidiagonalConjugationBand) = (ℵ₀,)
+@inline resizedata!(A::BidiagonalConjugationBand, n) = resizedata!(A.data, n)
+
+function _bcb_getindex(band::BidiagonalConjugationBand, I)
+    resizedata!(band, maximum(I) + 1)
+    if band.diag 
+        return band.data.dv[I]
+    else 
+        return band.data.ev[I]
+    end
+end
+
+@inline getindex(band::BidiagonalConjugationBand, I::Integer) = _bcb_getindex(band, I)
+@inline getindex(band::BidiagonalConjugationBand, I::AbstractVector) = _bcb_getindex(band, I)
+#@inline getindex(band::BidiagonalConjugationBand, I::AbstractInfUnitRange{<:Integer}) = view(band, I)
+#@inline getindex(band::SubArray{<:Any, 1, <:BidiagonalConjugationBand}, I::AbstractInfUnitRange{<:Integer}) = view(band, I)
+
+copy(band::BidiagonalConjugationBand) = band
+
+const BidiagonalConjugation{T} = Bidiagonal{T, BidiagonalConjugationBand{T}}
+
+"""
+    BidiagonalConjugation(U, X, V, uplo)
+
+Efficiently compute the projection of the matrix product
+`inv(U)XV` onto a bidiagonal matrix. The `uplo` argument 
+specifies whether the projection is upper (`uplo = 'U'`)
+or lower (`uplo = 'L'`) bidiagonal. 
+
+The computation is returned as a `Bidiagonal` matrix whose 
+diagonal and off-diagonal vectors are computed lazily.
+"""
+function BidiagonalConjugation(U, X, V, uplo)
+    _uplo = _to_uplo(uplo)
+    data = BidiagonalConjugationData(U, X, V, _uplo)
+    return _BidiagonalConjugation(data, _uplo)
+end
+
+function _BidiagonalConjugation(data, uplo) # need uplo argument so that we can take transposes
+    dv = BidiagonalConjugationBand(data, true)
+    ev = BidiagonalConjugationBand(data, false)
+    return Bidiagonal(dv, ev, uplo)
+end
+
+function copy(A::BidiagonalConjugation) # need a new method so that the data remains aliased between dv and ev 
```

Comment:
Good call! Though since the bands are not modifiable should we just map this a no-op? (`copy(A::BidiagonalConjugation) = A`)